### PR TITLE
[BUG] Fix NaN error for validating param sync

### PR DESCRIPTION
**Pull request by haolin-nju**
In validating parameter synchronization, comparing the src_tensor with dst_tensor by `torch.equal` will result in assertion error if NaN values presented. This PR bypasses such error by printing an warning message to the log and comparing the src_tensor with dst_tensor by `torch.allclose(.., rtol=0, atol=0, equal_nan=True)`.

### CODE REVIEW

**Review by SeaOfOcean — APPROVED** [review]

**Comment by SeaOfOcean** [comment]
LGTM

File: chatlearn/runtime/parameter_sync.py [hunk 1]
```diff
@@ -358,8 +359,19 @@ def validate():
                                                      recv_actor.get_parameter_to_sync.remote(dst_name, pipe_stage, True)])
                 assert src_tensor.shape == dst_tensor.shape, \
                     f"after weight sync {src_name}: {src_tensor.shape} and {dst_name}: {dst_tensor.shape} do not match"
-                assert (src_tensor == dst_tensor).all(), \
-                    f"after weight sync {src_name}: {src_tensor} and {dst_name}: {dst_tensor} do not match"
+                if not src_tensor.equal(dst_tensor):
+                    if src_tensor.isnan().any() or dst_tensor.isnan().any():
```
Comment (adoda) [inline chatlearn/runtime/parameter_sync.py:363]: 这里是不是要src和dst都是nan才绕过，否则一端是nan，一端不是，还是要报错把

Comment (haolin-nju) [reply chatlearn/runtime/parameter_sync.py:363]: allclose会在这种情况报错，不过改成and报错效率确实更高一点

Comment (haolin-nju) [reply chatlearn/runtime/parameter_sync.py]: done

**Review by adoda — APPROVED** [review]

**Comment by haolin-nju** [comment]
Close because the issue has been resolved in commit https://github.com/alibaba/ChatLearn/commit/e07e820df4877b3dc2c042ae6e263bce8228711f from PR https://github.com/alibaba/ChatLearn/pull/126.

**Pull request closed without merging**